examples/cloud-datafusion-functions-plugins/README.md

```diff
@@ -0,0 +1,374 @@
+
+**`CopyTableAction` plugin requires the following configuration:**
+
+-  Label: plugin label name.
+-  Key path: Service account key file path to call the Big Query API.
+-  Project ID: GCP project ID.
+-  Dataset: Big Query dataset name.
+-  Table Name: Big Query table name.
+
+**Please see the following screenshot for example:**
+
+![image](img/6-copy_table_action_ui.png)
+
+## `DropTableAction`
+
+**Plugin Description**  
+Drops a BigQuery table in the beginning of the pipeline runs.
+
+**Use Case**  
+Useful to drop staging tables.
+
+**Dependencies**  
+Requires BQ table to drop to exist.
+
+**Drop table action plugin requires the following configuration:**
+
+-  Label : plugin label name.
+-  Key path: Service account key file path to call the Big Query API.
+-  Project ID: GCP project ID.
+-  Dataset: Big Query dataset name.
+-  Table Name: Big Query table name.
+
+Please see the following screenshot for example configuration:
+
+![image](img/7-drop_table_action_ui.png)
+
+## `TruncateTableAction`
+
+**Plugin Description**  
+Truncates a BigQuery table when we set pipelines to restore the data from source.
+
+**Use Case**  
+Applicable in restoring data pipelines from source.
+
+**TruncateTable action plugin requires the following configuration:**
+
+-  Label : plugin label name.
+-  Key path: Service account key file path to call the Big Query API.
+-  Project ID: GCP project ID.
+-  Dataset: Big Query dataset name.
+-  Table Name: Big Query table name.
+
+**Please see the following screenshot for example configuration:**
+
+![image](img/8-truncate_table_action_ui.png)
+
+
+# Putting it all together into a Pipeline
+
+`CheckPointReadAction` → `TruncateTableAction` → Database → BigQuery → `MergeLastUpdateTSAction` → `CheckPointUpdateAction`
+
+What does the pipeline do?
+
+1. `CheckPointReadAction` - reads latest checkpoint from Firestore
+1. `TruncateTableAction` - truncate the records in the log table
+1. Database Source- imports data from the source
+1. BigQuery Sink - exports data into BigQuery from previous step (database source)
+1. `MergeLastUpdateTSAction` -  merge based on timestamp and the update column list (columns to keep in the merge).
+    -  Note: Alternatively, you can use [`BigQueryExecute`](https://github.com/data-integrations/google-cloud/blob/develop/src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java) action to do a Merge.
+1. CheckPointUpdateAction - update checkpoint in Firestore from the max record lastUpdateTimestamp in BigQuery 
+
+## Successful run of Incremental Pipeline
+
+![image](img/9-successful_run_incremental_v2_pipeline.png)
+
+### Runtime arguments (set latestWatermarkValue to 0)
+![image](img/9-set_runtime_arugments_latestWatermarkValue.png)
+
+### `CheckPointReadAction`
+
+**Label:**  
+CheckPointReadAction
+
+**Specify the document name to read the checkpoint details\*:**  
+INCREMENTAL_DEMO
+
+**Buffer time to add to checkpoint value. (Note: in Minutes):**  
+1
+
+**project:**  
```

Comment:
```suggestion
**`project`:**  
```
or 
```suggestion
**Project:**  
```
or 
```suggestion
**Project ID:**  
```

Reply:
I'm keeping this text as is as **project:**   is based on the screenshot below of the UI.